Adaptors/RabbitMQ/src/PullQueueStorage.cs

```diff
@@ -0,0 +1,197 @@
+                      {
+                        {
+                          "x-dead-letter-exchange", "ArmoniK.QueueExchange"
+                        },
+                        {
+                          "x-message-ttl", 100 // TODO: Make this a configurable variable?
+                        },
+                      };
+
+      var retryQueue = ConnectionRabbit.Channel!.QueueDeclare("",
+                                                              false,
+                                                              true,
+                                                              false,
+                                                              retryArgs);
+
+      ConnectionRabbit.Channel.QueueBind(retryQueue.QueueName,
+                                         "ArmoniK.RetryExchange",
+                                         Options!.PartitionId);
+
+      /* Setup prefetching. TODO: Rename LinkCredit to something less amqpLite specific */
+      ConnectionRabbit.Channel.BasicQos(0,
+                                        Convert.ToUInt16(Options.LinkCredit),
+                                        false);
+
+      var consumer = new AsyncEventingBasicConsumer(ConnectionRabbit.Channel!);
+
+      // Delegate to declare a subscriber to the queue
+      Task Subscriber(object?               model,
+                      BasicDeliverEventArgs eventArgs)
+      {
+        var body    = eventArgs.Body.ToArray();
+        var message = Encoding.UTF8.GetString(body);
+
+        // Enqueue message in local storage to be pulled later, the
+        // Cancellation token will be set by the pulling function.
+        queueMessageHandlers_.Enqueue(new QueueMessageHandler(ConnectionRabbit.Channel!,
+                                                              eventArgs,
+                                                              message,
+                                                              logger_,
+                                                              cancellationToken));
+        return Task.CompletedTask;
+      }
+
+      consumer.Received += Subscriber;
+
+      ConnectionRabbit.Channel!.BasicConsume(pullQueue.QueueName,
+                                             false,
+                                             consumer);
+      IsInitialized = true;
+    }
+  }
+
+  public async IAsyncEnumerable<IQueueMessageHandler> PullMessagesAsync(int                                        nbMessages,
+                                                                        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+  {
+    var nbPulledMessage = 0;
+
+    if (!IsInitialized)
+    {
+      throw new ArmoniKException($"{nameof(PullQueueStorage)} should be initialized before calling this method.");
+    }
+
+    // TODO: Fix this, I put it here cuz the method must be async to return an IAsyncEnumerable.
+    await Task.CompletedTask.ConfigureAwait(false);
```

Comment:
Maybe try ValueTask that are lighter than Task. Or juste suppress the warning

Reply:
Ok; I'll give it a try